internal/provider/namespace_resource.go

```diff
@@ -396,9 +410,15 @@ func (r *namespaceResource) Update(ctx context.Context, req resource.UpdateReque
 		resp.Diagnostics.AddError("Failed to get current namespace status", err.Error())
 		return
 	}
-	codecServer := getCodecServerFromModel(ctx, resp.Diagnostics, &plan)
-	if resp.Diagnostics.HasError() {
-		return
+
+	var codecServer *namespacev1.CodecServerSpec
+	if !plan.CodecServer.IsNull() {
```

Comment:
This bug was uncovered by the acceptance tests after fixing the diagnostic errors